hub/auth/internal/browser/open.go

```diff
@@ -17,6 +17,8 @@ const (
 	switchPath string = "tenants"
 )
 
+// OpenBrowserForSwitch opens the default web browser to the tenant switching page for the given AuthConfig.
+// If a tenant name is provided, it will be preselected in the UI.
 func OpenBrowserForSwitch(authConfig *sessionstore.AuthConfig, tenant ...string) error {
```

Comment:
Hi @tkircsi, is it possible to make the auth-config reusable for other IDP providers. It would be useful to make it generic enough so that we can reuse it in other places beyond Hub (for example signing). We just need to make it generic and avoid coupling with Hub-specific details. These can be done as part of the request instead of being baked into the logic itself.